### PR TITLE
Do not present local notifications if a read receipt arrives immediat…

### PR DESCRIPTION
**Pull request by charlesmchen**
…ely after.

This fixes: "my phone constantly vibrates while I'm chatting on desktop."

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/network/PushManager.m [hunk 1]
```diff
@@ -429,19 +431,61 @@ - (BOOL)applicationIsActive {
...
+{
+    OWSAssert([NSThread isMainThread]);
+
+    NSString *threadId = notification.userInfo[Signal_Thread_UserInfo_Key];
+    if (checkForCancel && threadId != nil) {
+        [_pendingNotifications addObject:notification];
+
+        // The longer we wait, the more obsolete notifications we can suppress -
+        // but the more lag we introduce to notification delivery.
+        const CGFloat kDelaySeconds = 0.3f;
```
Comment (charlesmchen) [inline Signal/src/network/PushManager.m]: A key line.

**Comment by charlesmchen** [comment]
I don't understand how this issue doesn't affect more people on the team.

**Comment by michaelkirk** [comment]
@charlesmchen what's the repro for this behavior?

Is it...

I am Alice, and I am linked with Desktop.
When I send messages to Bob from Desktop, my phone vibrates

Is that right?

File: Signal/src/network/PushManager.m (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/network/PushManager.m]: Would it be simpler to modify the fireDate?

e.g. replace the entirety of this method implementation with:

```
- (void)presentNotification:(UILocalNotification *)notification checkForCancel:(BOOL)checkForCancel
{
    AssertIsOnMainThread();
    if (checkForCancel && threadId != nil) {
        _pendingNotification.fireDate = [NSDate dateWithTimeIntervalSinceNow:0.3f];
        _pendingNoficitation.timeZone = [NSTimeZone localTimeZone]; // IIRC this is necessary.
    }
    [[UIApplication sharedApplication] scheduleLocalNotification:notification];
    [self.currentNotifications addObject:notification];
}
```

Then most of the other changes could be omitted too.

Comment (charlesmchen) [inline Signal/src/network/PushManager.m]: Marvelous, thank you.

**Pull request merged**